include/revng/Pipeline/Invokable.h

```diff
@@ -0,0 +1,278 @@
+#pragma once
```

Comment:
I assume this file is mostly a refactoring of existing stuff. Please point me to some point you'd like to get feedback on, if any.

Reply:
the only weird thing is that there is `InvokableImpl` that implements `InvokableBase`. `InvokableImpl` is used as a memeber of `PipeImpl` and then `PipeImpl` reimplements `InvokableBase` by dispatching the call to the field.